rustls/src/conn.rs

```diff
@@ -62,7 +62,8 @@ impl IoState {
 
 /// A structure that implements `std::io::Read` for reading plaintext.
 pub struct Reader<'a> {
-    common: &'a mut ConnectionCommon,
+    received_plaintext: &'a mut ChunkVecBuffer,
```

Comment:
The immediate reason was this to prevent the type argument from `ConnectionCommon` having to leak into the `Connection` trait as an associated type, which would preclude turning `Connection` instances into trait objects, but I also like that this moves the logic out of the generic `ConnectionCommon` type and into the specific `Reader` where it is actually used.